src/main/java/ch/njol/skript/expressions/ExprPotionEffectTier.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Potion Effect Tier")
+@Description("An expression to obtain the amplifier of a potion effect applied to an entity.")
+@Examples("if the amplifier of haste of player >= 3:")
+@Since("INSERT VERSION")
+public class ExprPotionEffectTier extends SimpleExpression<Number> {
+
+	static {
+		Skript.registerExpression(ExprPotionEffectTier.class, Number.class, ExpressionType.COMBINED,
+			"[the] [potion] (tier|amplifier|level) of %potioneffecttype% (of|for) %livingentity%"
```

Comment:
I think (of|for|on) would be a good addition to the syntax. Also this syntax could support checking multiple types and multiple entities.

Reply:
I like the addition to the syntax tbh, I'll add that:) but we discussed adding multiple types and entities but the result would just be a bit messy, https://github.com/SkriptLang/Skript/pull/4632#discussion_r922610737, see here.

Reply:
There are multiple ways you can do it. Majority of Skript's syntaxes support multiple values and with my pull request at https://github.com/SkriptLang/Skript/pull/4898 doing this will be even more simpler.

Here is an example of doing it using Java streams
```java
@Override
@Nullable
protected Integer[] get(Event event) {
	return entities.stream(event)
		.flatMap(entity -> types.stream(event)
				.map(type -> entity.getPotionEffect(type))
				.map(potion -> potion == null ? 0 : potion.getAmplifier() + 1))
		.toArray(Integer[]::new);
}
```

After my pull request
```java
@Override
@Nullable
protected Integer[] get(Event event) {
	return entities.get(event, entity -> types.get(event, type -> {
		PotionEffect potion = entity.getPotionEffect(type);
		return potion == null ? 0 : potion.getAmplifier() + 1;
	}));
}
```

Reply:
I see how it can be done, however it just seems like the result would be messy and no way to really tell, I guess I could give it a shot but it just seems pointless.

Reply:
Supporting multiple in the expression is a basic requirement for any syntax.

Reply:
I know, just seems messy for the user, ill add it when I have a min